server/container_update_resources.go

```diff
@@ -51,3 +56,57 @@ func toOCIResources(r *pb.LinuxContainerResources) *rspec.LinuxResources {
 		// TODO(runcom): OOMScoreAdj is missing
 	}
 }
+
+func (s *Server) updateContainerResources(ctx context.Context, c *oci.Container, resources *rspec.LinuxResources) (retErr error) {
+	if err := updateContainerSpec(c, resources); err != nil {
+		return err
+	}
+
+	defer func() {
+		if retErr != nil {
+			oldSpec := c.Spec()
+			oldSpecGen := generate.Generator{Config: &oldSpec}
+			saveOptions := generate.ExportOptions{}
+			if err := oldSpecGen.SaveToFile(filepath.Join(c.BundlePath(), "config.json"), saveOptions); err != nil {
```

Comment:
uhm, AFAIU we should not do this: https://github.com/cri-o/cri-o/issues/3963#issuecomment-659629338

Reply:
I do not see another option, I checked the contained code, and it does the same thing.
@haircommander @mrunalp Guys, do you have a better option how can I achieve the update of the container spec?